### PR TITLE
Remove member equality judgments.

### LINKED ISSUE #457: Remove member equality judgments
Now that we have equality pretypes, it is technically possible for us to remove member equality judgments and work in the Nuprl style.

Related to #351.

### PR DESCRIPTION
**Pull request by favonia**
This shows how to close #457. Some adjustments to `reduce` are needed.

### CODE REVIEW

**Comment by favonia** [comment]
The remaining issue is #382... how should we design the "`deep-reduce`"? If the conclusion is `(= ty m n)`, and the user wants to reduce `m`, what should the user do? One proposal:

1. `reduce in concl at left`

The issue is that it is unclear what `reduce in x y concl at left right` means, if we want to allow reducing at multiple spots. Or, we can just allow the following two tactics:

1. `reduce [x | concl]+`
2. `reduce [x | concl] at [left | right | type]+`

Thoughts?

**Comment by jonsterling** [comment]
@favonia It seems like `reduce` should just take a bunch of `selector,accessor` pairs somehow. Do you think that would resolve the problem?

**Comment by favonia** [comment]
The problem is the grammar design...

**Comment by favonia** [comment]
How would you want to write down "reduce both sides of x (which is an EQ_TYPE) and the left side of the conclusion (which is an internalized EQ)"?

**Comment by jonsterling** [comment]
What about `reduce x at left right, concl at left`?

**Comment by favonia** [comment]
Well, I think we already removed commas...

**Comment by jonsterling** [comment]
@favonia You are right...

And I'm sorry about the merge conflicts, I didn't think of that.

**Comment by favonia** [comment]
Any other ideas? (Don't worry. The merge conflict is easy to solve.)

**Comment by jonsterling** [comment]
> Any other ideas?

Well, my go-to solution for this kind of problem is to add more brackets or something, but that's not really ideal :laughing: 

Or we could have a notation like `reduce ((x|concl).(left|right|type))+` or something. In particular, your example would be written as follows:

```
reduce x.left x.right concl.left
```

or we could even have

```
reduce x.[left right] concl.left
```

if you felt that was better.

**Comment by favonia** [comment]
@jonsterling Alright I implemented the proposed rewrite in #528.

**Comment by favonia** [comment]
@jonsterling I am worried about periods being confusing... did we remove all other usages of periods?

**Comment by favonia** [comment]
Hmm. Unfinished goals in `test/success/bool-fhcom-without-open-eval.prl` are very suspicious. They can solved by `auto` but `lam` somehow fails to solve them?

**Comment by favonia** [comment]
Same things in `test/success/num.prl` now. `lam` is leaving goals that can be discharged by `auto`.

**Comment by favonia** [comment]
Same in `test/success/universes.prl`. @jonsterling Is this something expected due to recent changes?

**Comment by jonsterling** [comment]
@favonia This reminds me of a long-standing issue with `lam`, that we have the ability to solve using https://github.com/RedPRL/sml-dependent-lcf/pull/41/files

**Comment by favonia** [comment]
Did I create too many aux goals in `Fun.True` or something?

File: test/success/V-types.prl [hunk 1]
```diff
@@ -73,7 +73,7 @@ Thm IdV/Test2(#l:lvl) : [
    (= ty (coe 0~>1 [x] (IdV x #l ty) a)
          (coe 0~>1 [_] ty a)))
 ] by [
-  lam ty a => auto
+  lam ty a => reduce at left; auto
```
Comment (favonia) [inline test/success/V-types.prl]: I wonder why this is needed? Somehow `ComputeBefore` stopped working.

**Comment by favonia** [comment]
@jonsterling Strangely, by fixing some sort bugs, some lingering subgoals after `lam` disappeared. However, I do not feel comfortable with this kind of magical fix. It seems something deeper is still wrong. I might put in extra `auto` for now.

File: example/semi-simplicial.prl [hunk 2]
```diff
@@ -82,7 +82,7 @@ Thm Pick/compose : [
         ]
       ]
     ]
-  ]
+  ]; auto
```
Comment (favonia) [inline example/semi-simplicial.prl]: Ugly hack here.

File: test/success/V-types.prl (hunk 1 above)
Comment (favonia) [reply test/success/V-types.prl]: This was due to a bug in `AtomicJudgment` which is now fixed.

**Pull request merged**